module/apmazure/storage.go

```diff
@@ -0,0 +1,128 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build go1.14
+// +build go1.14
+
+package apmazure // import "go.elastic.co/apm/module/apmazure"
+
+import (
+	"context"
+	"errors"
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/Azure/azure-pipeline-go/pipeline"
+
+	"go.elastic.co/apm"
+	"go.elastic.co/apm/module/apmhttp"
+	"go.elastic.co/apm/stacktrace"
+)
+
+func init() {
+	stacktrace.RegisterLibraryPackage(
+		"github.com/Azure/azure-storage-blob-go/azblob",
+	)
+}
+
+// WrapPipeline wraps the provided pipeline.Pipeline, returning a new one that
+// instruments requests and responses.
+func WrapPipeline(p pipeline.Pipeline) pipeline.Pipeline {
+	return &apmPipeline{p}
+}
+
+type apmPipeline struct {
+	next pipeline.Pipeline
+}
+
+func (p *apmPipeline) Do(
+	ctx context.Context,
+	methodFactory pipeline.Factory,
+	req pipeline.Request,
+) (pipeline.Response, error) {
+	rpc, err := newAzureRPC(req)
+	if err != nil {
+		return p.next.Do(ctx, methodFactory, req)
+	}
+
+	tx := apm.TransactionFromContext(ctx)
+	if tx == nil {
+		return p.next.Do(ctx, methodFactory, req)
+	}
+
+	span := tx.StartSpan(rpc.name(), rpc._type(), apm.SpanFromContext(ctx))
```

Comment:
```suggestion
	span := apm.StartSpan(ctx, rpc.name(), rpc._type())
```

If ctx doesn't contain a transaction, this will return a dropped span. If it does, it will also check if ctx contains a span and use that as the parent.

Reply:
That was my understanding too, but the tracing doc has it marked as "required": https://github.com/elastic/apm/blob/master/specs/agents/tracing-instrumentation-azure.md#span-context-fields. I raised the question in slack but didn't receive a response.